### DISCUSSION TITLE
RAN metrics

### DISCUSSION THREAD

**Discussion by altis82** [srsran/srsRAN_Project#328] [General Help] [answered]
Hi everyone,
I would like to save these metrics in the csv file
![image](https://github.com/srsran/srsRAN_Project/assets/6227022/3c17d448-811e-4839-b585-28cad77f61fc)
Do you have any idea how to do? Thanks

**Comment by brendan-mcauliffe** [comment, marked as answer]
Hi @altis82, we do not support a `.csv` output for the RAN metrics. We do have support for outputting these as a `.json` file. Check the `metrics` section of the configuration file reference [here](https://docs.srsran.com/projects/project/en/latest/user_manuals/source/config_ref.html).  

There are many tools that will convert this to `.csv`.

**Comment by altis82** [comment]
Yes, I could enable to get that json file but I see sometimes the parameters are not correct. As you can see, at this moment the UE is connected but it shows some weird numbers. I would like to go customize and add more parameters
![image](https://github.com/srsran/srsRAN_Project/assets/6227022/8705146b-895d-4383-b714-1862f0f5972e)

**Reply by brendan-mcauliffe** [reply]
Can you make sure you are using the latest version of the code?

**Reply by altis82** [reply]
yes, you are right. I am using eNB

**Reply by brendan-mcauliffe** [reply]
You should get an output similar to the following for the srsRAN Project gNB. 

```
{
  "timestamp": 1700671417.005,
  "ue_list": [
    {
      "ue_container": {
        "pci": 1,
        "rnti": 17921,
        "cqi": 15,
        "ri": 1,
        "dl_mcs": 27,
        "dl_brate": 291680.0,
        "dl_nof_ok": 352,
        "dl_nof_nok": 0,
        "dl_bs": 0,
        "pusch_snr_db": 24.025097,
        "ul_mcs": 26,
        "ul_brate": 12838928.0,
        "ul_nof_ok": 431,
        "ul_nof_nok": 0,
        "bsr": 0
      }
    }
  ]
}
```
This is a taken from a set-up using srsUE the srsRAN Project gNB and ZMQ. The traffic is generated using iPerf.

**Reply by s21sm** [reply]
@brendan-mcauliffe Could you please instruct how I can get this json file? I enabled the 

```
metrics: 
  rlc_json_enable: 1                # Enable RLC metrics reporting (need to deliver measurements to E2SM_KPM service model)
  rlc_report_period: 1000           # Set reporting period to 1s
```

**Reply by pgawlowicz** [reply]
@s21sm you can try with this solution: https://github.com/srsran/srsRAN_Project/discussions/539#discussioncomment-9008517

you need to add in metrics section:
```
  addr: 127.0.0.1                     # Optional TEXT:IPV4. Sets the metrics address. Supported: IPV4 address. 
  port: 55555                           # Optional UINT. Sets the metrics UPD port. Supported: [0 - 65535].
```
